rootston/output.c

```diff
@@ -695,6 +733,14 @@ static void damage_from_surface(struct wlr_surface *surface,
 	}
 }
 
+void output_damage_from_local_surface(struct roots_output *output,
+		struct wlr_surface *surface, double ox, double oy, float rotation) {
+	struct wlr_output_layout_output *layout = wlr_output_layout_get(
+		output->desktop->layout, output->wlr_output);
+	damage_from_surface(surface, ox + layout->x, oy + layout->y,
+			rotation, output);
```

Comment:
Should use `surface_for_each_surface` to apply damage from subsurfaces too.

Reply:
Doing subsurfaces later, added a todo